components/raftstore/src/store/peer.rs

```diff
@@ -1747,7 +1898,26 @@ where
 
         let mut ready = self.raft_group.ready();
 
-        self.last_unpersisted_number = ready.number();
+        let mut proposal_times = vec![];
+
+        if ctx.raft_metrics.waterfall_metrics {
```

Comment:
Create another PR for new Grafana panels later?

Reply:
Yes, I plan to introduce `waterfall_metrics` in another PR. By the way, https://github.com/tikv/tikv/pull/10326 is split from this PR for introducing `async_io.rs`.